soc/xtensa/intel_adsp/common/ace/power_down.S

```diff
@@ -0,0 +1,159 @@
+/* Copyright (c) 2022 Intel Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+ #include "asm_memory_management.h"
+
+	.section .text, "ax"
+	.align 64
+power_down_literals:
+	.literal_position
+ipc_flag:
+	.word 0x80000000 // IPC_DIPCTDR_BUSY
+sram_dis_loop_cnt:
+	.word 4096
+
+	.global ace_power_down
+	.type ace_power_down, @function
+
+ /**
+ * @brief Perform power down.
+ *
+ * Depending on arguments, memories are switched off.
+ *
+ * @param A2 - argument for LPSRAM
+ * @param A3 - pointer to array containing power gating mask.
+ * Size of array is determined by MAX_MEMORY_SEGMENTS define.
+ * @param A4 - send response to ipc
+ */
+
+#define IPC_HOST_BASE                0x00073000
+#define b_enable_lpsram              a2
+#define pu32_hpsram_mask             a3
+#define b_ipc_response               a4
+#define temp_reg0                    a6
+#define temp_reg1                    a7
+#define temp_reg2                    a8
+#define temp_reg3                    a9
+#define temp_reg4                    a10
+#define temp_reg5                    a11
+#define temp_reg6                    a12
+#define p_ipc_regs                   a13
+#define u32_ipc_response_mask        a14
+#define pfl_reg                      a15
+
+ace_power_down:
```

Comment:
To repeat from earlier: I can't see how we can reach ace_power_down() from the code as contributed.  Is this not dead code?  How do we exercise?

Reply:
@andyross 
file: multiprocessing.c
function: pm_state_set
line: 165

If your concerns are about that it will not be executed without **\"_CONFIG_PM=y_\"**, thats true. Unfortunately, no one has raised such objections before.

Reply:
This is optional and applications enable / disable it as they want. The majority of boards supporting PM don't enable it by default.